_videos/2017-01-30-complex-refactoring-in-simple-steps-III.md

```diff
@@ -0,0 +1,21 @@
+---
+author: Matthew Butt
+layout: video
+name: complex-refactoring-in-simple-steps-III
+image:
+ src: /assets/img/custom/videos/complex-refactoring-in-simple-steps-III.jpg
+
+title: "Complex refactoring in simple steps Part III"
+hidden: true
+video-url: https://www.youtube.com/embed/SuqrAtXnn1U
+date: 2017-01-30 10:00:00 +00:00
+---
+
+
+## Screencast - Complex refactoring in simple steps, Part III From Conditional to Polymorphism
+
+Matthew Butt demonstrates how to compose simple automated refactoring steps in ReSharper to refactor code to better designs.
+
+In this episode we identify three classes that are being used in the same way. We introduce a common abstraction, and separate their creation and invocation. Finally, we remove a Middleman class to complete our journey to polymorphism.
```

Comment:
Nice suggestion. How about this: https://sourcemaking.com/refactoring/smells/middle-man ?